Formula/l/llvm.rb

```diff
@@ -40,12 +29,10 @@ class Llvm < Formula
   keg_only :provided_by_macos
 
   # https://llvm.org/docs/GettingStarted.html#requirement
-  # We intentionally use Make instead of Ninja.
-  # See: Homebrew/homebrew-core/issues/35513
```

Comment:
This issue, reported over 4 years ago, was with LLVM 7. We can probably consider switching back to Ninja after 8 major releases. As tested locally on Linux this also helped to fix the parallelisation issue below.

Reply:
Seems to work: https://github.com/Homebrew/homebrew-core/actions/runs/6137360618/job/16653145371

Reply:
Original concern with Ninja was bottle size, so worth comparing that when CI finishes.

Reply:
The bottle sizes look pretty reasonable:

```console
$ gzip -l llvm--17.0.0-rc4.*.bottle.tar.gz
         compressed        uncompressed  ratio uncompressed_name
          446947058          1782855680  74.9% llvm--17.0.0-rc4.arm64_big_sur.bottle.tar
          447281041          1781954560  74.9% llvm--17.0.0-rc4.arm64_monterey.bottle.tar
          449510988          1790627840  74.9% llvm--17.0.0-rc4.arm64_ventura.bottle.tar
          471152768          1878732800  74.9% llvm--17.0.0-rc4.big_sur.bottle.tar
          471629802          1880043520  74.9% llvm--17.0.0-rc4.monterey.bottle.tar
          473800535          1886115840  74.9% llvm--17.0.0-rc4.ventura.bottle.tar
          570711687          2565222400  77.8% llvm--17.0.0-rc4.x86_64_linux.bottle.tar
         3331033879         13565552640  75.4% (totals)
```

That's around 1.8GB for macOS, 2.5GB for Linux.

In comparison:

```console
(x86_64 Ventura) $ du -sh "$(realpath "$(brew --prefix llvm)")"
1.6G	/usr/local/Cellar/llvm/16.0.6
(x86_64 Linux) $ du -sh "$(realpath "$(brew --prefix llvm)")"
2.1G	/home/linuxbrew/.linuxbrew/Cellar/llvm/16.0.6
```

Reply:
I'm not sure I understand. By "risk" did you mean there are still concerns about bottle sizes? If we don't switch to Ninja in this PR, the `llvm` and `llvm@16` formulae wouldn't have parallel builds on Linux.

Reply:
What I mean by "avoid increasing risk" is "change less things at once". This PR will instantly change the meaning to `brew install llvm` in two ways (cmake -> ninja, 16 -> 17) rather than just one. It seems better to disentangle the two and presumably changing just cmake -> ninja can land without all the other changes pinning formulae to llvm@16.

Reply:
> What I mean by "avoid increasing risk" is "change fewer things at once".

All the changes to the `llvm` formula have already been tested on the final release candidate, so it's very likely that they still work on the 17.0.1 release. It's not that we can't make fewer changes in this PR; it's just less practical to do so considering that a single build-test cycle for `llvm` can last for more than 24 hours. (And in the midst of mass bottling, CI resources are really valuable to us.)

> This PR will instantly change the meaning to `brew install llvm` in two ways (cmake -> ninja, 16 -> 17) rather than just one.

Please note that the switch to Ninja only affects source builds; nothing extra will be installed with `brew install llvm` after this PR.

> without all the other changes pinning formulae to llvm@16

These are also necessary changes because they are dependents of `llvm` and they *have to* be rebuilt on LLVM major version bumps. We are already speeding up the progress of this PR by not migrating any of them to the latest `llvm` 17 here.